src/main/java/com/minecolonies/coremod/client/gui/modules/DOCraftingWindow.java

```diff
@@ -103,18 +103,8 @@ private void addRecipe()
             }
         }
 
-        int inputCount = 0;
-        for (int i = 0; i < inputInventory.getContainerSize(); i++)
-        {
-            if (!inputInventory.getItem(i).isEmpty())
-            {
-                inputCount++;
-            }
-        }
-
-        final List<Integer> inputIndizes = map.get(inputCount);
         final List<ItemStorage> input = new ArrayList<>();
-        for (int i = 0; i < inputCount; i++)
+        for (int i = 0; i < inputInventory.getContainerSize(); ++i)
```

Comment:
I'm not sure this works reliably. I remember that I added this on purpose because else, in some situations some things wouldn't work like:
- Having multiple items in the input, but some of the recipe subsets only require a subset of those items, etc.

Reply:
I changed it because the previous version was not correct if there were any empty items (although there probably shouldn't have been, which is why it mostly worked).

It first looped through the entire inventory, counting how many items were non-empty (N).  Then it looped through only the first N slots of the inventory, again ignoring empty items, but also ignoring any non-empty >N slots.

This code only ever added one recipe, it didn't support subsets.  The UI is the same -- when you put in one slot it will show a bunch of panels, but these go away once you fill in two slots.

It looked ok when I tested it, but if there's a specific combo you think might be problematic, I can try it out and see if it works.

Reply:
Having said that, I'm seeing a different bug while testing this further (not related to these changes, it happens in a clean build too): when I add a recipe to a building it plays the happy noise but does not appear in the recipe list until I relog.

Reply:
Isn't container size always 3? And what happens if you add/remove inputs?

Reply:
There's still an empty slot check, so it still calculates `inputCount` (though here now `input.size()`) correctly.  I just merged the two loops you had into one, which avoided a potential bug.

Reply:
ahk